### DISCUSSION TITLE
use jsdoc to generate json schema options

### DISCUSSION THREAD

**Discussion by xddq** [sinclairzx81/typebox-codegen#13] [Ideas]
Hey @sinclairzx81,

what do you think about the idea to use jsdoc in the typescript to typebox part of this package to generate JSON schema options?

I started with the implementation for jsdoc for the number type here https://github.com/sinclairzx81/typebox-codegen/pull/12

**Comment by sinclairzx81** [comment]
@xddq Hi,

Yeah, could be keen on this :) The problem though (afaik) is that it's not possible to get jsdoc comments for flat types, but it's been years since I've looked into this (it may be possible now). Would want to be able to support the following.
```typescript
/**
 * @minimum 10
 * @maximum 20
 * @multipleOf 2
 */
type T = number // I'm not sure TS supports JS docs for non `interface` types.
```
I've added some very rough hooks in there for `option` injection on the types in the last updates (it needs work tbh), but it should be possible to inject options on the types if you can extract them from jsdoc comments. I don't think the properties need to be strict (it should support known and unknown property types), with the values assigned as `SchemaOptions`.

```typescript
interface Foo {
  /**
   * @minimum 100
   * @minimum 200
   * @multipleOf 2
   * @default 150
   * @description "it's a number" - strings must be quoted
   * @foobar "should support unknown props"
   */
  x: number
}

const Foo = Type.Object({
  x: Type.Number({
     minimum: 100,
     maximum: 200,
     multipleOf: 2,
     default: 150,
     description: "it's a number",
     foobar: "should support unknown props"
  })
})
```

Might also need to be able to support objects on the jsdoc comments. 
Open to thoughts!
S

**Reply by xddq** [reply]
Uuh, this hook seems to be all I need, nice! Alright, I will see what I can do :]

I can confirm that typescript picks up that jsdoc comment exists for flat types (that was the one example I did implement and test already).

I like the idea with forcing "" and parsing the rest to number (or perhaps number | bigint). I did use doctrine, will have a look if the extra "" end up in the parsed token.
